File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const yearningByName = {}
    const answer = []
    
    for (let i = 0; i < name.length; i++) {
        yearningByName[name[i]] = yearning[i]
    }
    
    for (let i = 0; i < photo.length; i++) {
        let sum = 0;
        for (let j = 0; j < photo[i].length; j++) {
            if (yearningByName[photo[i][j]]) {
                sum += yearningByName[photo[i][j]];
            } else {
                sum += 0;
            }
        }
        answer.push(sum);
    }
    
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    score_map = {n: y for n, y in zip(name, yearning)}
    answer = []
    
    for group in photo:
        total = 0
        for person in group:
            total += score_map.get(person, 0)
        answer.append(total)
        
    return answer